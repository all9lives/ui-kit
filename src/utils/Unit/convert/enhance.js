import R from 'ramda'

export const ignorePercentage = R.unless(R.contains('%'), R.__)
export const ignoreAuto = R.unless(R.contains('auto'), R.__)

export default {
  ignorePercentage,
  ignoreAuto
}
