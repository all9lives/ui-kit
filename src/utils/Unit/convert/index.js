export default unit => `${unit / 16}rem`
