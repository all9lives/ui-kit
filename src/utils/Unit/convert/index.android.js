import { identity } from 'ramda'
import { ignorePercentage, ignoreAuto } from './enhance'

const transform = value => `${value}px`

export default (unit, converter = identity) => ignoreAuto(ignorePercentage)(transform)(converter(unit))
