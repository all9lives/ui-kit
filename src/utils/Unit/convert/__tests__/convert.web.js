/* eslint-env jest */
import convert from '../index.js'

describe('Unit.convert.web', () => {
  it('converts numbers to values in `rem` unit', () => {
    expect(convert(16)).toMatchSnapshot()
  })

  it('does not convert percentage value', () => {
    expect(convert('10%')).toMatchSnapshot()
  })
})
