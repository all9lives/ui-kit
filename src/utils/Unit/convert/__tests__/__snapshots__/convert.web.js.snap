// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit.convert.web converts numbers to values in \`rem\` unit 1`] = `"1rem"`;

exports[`Unit.convert.web does not convert percentage value 1`] = `"10%"`;
